lsm-srlinux/ci/Jenkinsfile

```
@@ -119,35 +132,29 @@ pipeline {
             steps {
                 script {
                     withCredentials([
-                        // Credentials to access the containers repository on Cloudsmith.
-                        usernamePassword(
-                            credentialsId: "cloudsmith-docker-token",
-                            usernameVariable: "CLOUDSMITH_DOCKER_USERNAME",
-                            passwordVariable: "CLOUDSMITH_DOCKER_TOKEN"
-                        ),
                         // Credentials to access the inmanta-service-orchestrator-* repositories on Cloudsmith.
                         string(
                             credentialsId: "cloudsmith-token",
                             variable: "CLOUDSMITH_PKGS_TOKEN",
                         ),
                     ]) {
-
+                        // update_topology validates whether the given version has a valid format
                         update_topology(env.version)
                         update_project_yml(env.version, env.CLOUDSMITH_PKGS_TOKEN)
                         sh '''
                             cd "${WORKSPACE}/lsm-srlinux"
-                            echo "${CLOUDSMITH_DOCKER_TOKEN}" | sudo docker login -u "${CLOUDSMITH_DOCKER_USERNAME}" --password-stdin containers.inmanta.com
                             # Ensure we have the latest version of the srlinux image
                             sudo docker pull ghcr.io/nokia/srlinux:latest
 
                             # Create python environment
                             python3 -m venv venv
                             source "venv/bin/activate"
                             pip install -U pip
-                            if [ -n "${version}" ]; then
-                                pip install --pre pytest-inmanta-extensions "inmanta-service-orchestrator==${version}"
+                            if [[ ${version} == *dev ]]; then
+                                major_version=${version::-3}
+                                pip install --pre "inmanta-core[pytest-inmanta-extensions]" "inmanta-service-orchestrator~=${major_version}.0"
```

```suggestion
                                pip install "inmanta-core[pytest-inmanta-extensions]" "inmanta-service-orchestrator~=${major_version}.0.dev"
```
I think this is safer: it's what we want and nothing more. `--pre` allows everything (including external) packages to be a pre-release. I think we should avoid it whenever possible.